# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: xpcom/io/nsLocalFileWin.cpp
# Commit: ef46ec074d6a
# Full Hash: ef46ec074d6aba10d6e4342e305f36ead6996ddb
# Author: Honza Bambas <honzab.moz@firemni.cz>
# Date: 2018-06-07 21:50:48
# Regressor Bug: 1413868
# File Overlap Count: 1
# Description:
#   Bug 1413868. r=valentin
# ==============================================================================

diff -r 14f7dab3a9bd -r ef46ec074d6a xpcom/io/nsLocalFileWin.cpp
--- a/xpcom/io/nsLocalFileWin.cpp	Wed Jun 06 14:50:17 2018 -0700
+++ b/xpcom/io/nsLocalFileWin.cpp	Thu Jun 07 13:56:16 2018 -0400
@@ -42,6 +42,7 @@
 #include "prproces.h"
 #include "prlink.h"
 
+#include "mozilla/FilePreferences.h"
 #include "mozilla/Mutex.h"
 #include "SpecialSystemDirectory.h"
 
@@ -1049,6 +1050,10 @@
     return NS_ERROR_FILE_UNRECOGNIZED_PATH;
   }
 
+  if (FilePreferences::IsBlockedUNCPath(aFilePath)) {
+    return NS_ERROR_FILE_ACCESS_DENIED;
+  }
+
   if (secondChar != L':' && (secondChar != L'\\' || firstChar != L'\\')) {
     return NS_ERROR_FILE_UNRECOGNIZED_PATH;
   }
@@ -1803,6 +1808,10 @@
     dwCopyFlags |= COPY_FILE_NO_BUFFERING;
   }
 
+  if (FilePreferences::IsBlockedUNCPath(destPath)) {
+    return NS_ERROR_FILE_ACCESS_DENIED;
+  }
+
   if (!move) {
     copyOK = ::CopyFileExW(filePath.get(), destPath.get(), nullptr,
                            nullptr, nullptr, dwCopyFlags);